pkg/apis/core/validation/validation.go

```diff
@@ -3551,11 +3551,28 @@ func ValidatePreferredSchedulingTerms(terms []core.PreferredSchedulingTerm, fldP
 	return allErrs
 }
 
+func validatePodAffinityTermNamespaceSelector(podAffinityTerm core.PodAffinityTerm, fldPath *field.Path) field.ErrorList {
+	nsSelector := podAffinityTerm.NamespaceSelector
+	if nsSelector == nil {
+		return nil
+	}
+
+	allErrs := unversionedvalidation.ValidateLabelSelector(nsSelector, fldPath)
```

Comment:
I think, maybe a bit captious, `ValidateLabelSelector` applies a larger scope. So if nsSelector.matchExpressions happens to have non-`In` operator, and its values happens to be problematic, we may end up getting unnecessay validation errors.

For example, matchExpressions[0].Operator == Exists, and its values is not nil. We will get two errors:

1. values must not be specified when operator is Exists
2. operator must be IN

My option is that the first error is redundant.

Reply:
I thought about this as well, I think we should validate the valid operator first then do the generic validation.